### PR TITLE
fix: show deprecation warning for incorrect usage of Node.js API

### PR DESCRIPTION
**Pull request by snitin315**
- [x] This is a **bugfix**
- [ ] This is a **feature**
- [ ] This is a **code refactor**
- [x] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?
WIP ON TESTS

### Motivation / Use-Case

SHOW DEPRECATION WARNING

### Breaking Changes
NO

### Additional Info
NO

### CODE REVIEW

File: lib/Server.js [hunk 1]
```diff
@@ -20,6 +20,13 @@ class Server {
     // TODO: remove this after plugin support is published
     if (options.hooks) {
       [options, compiler] = [compiler, options];
+
+      this.compiler = compiler;
+      this.logger = this.compiler.getInfrastructureLogger("webpack-dev-server");
```
Comment (alexander-akait) [inline lib/Server.js:25]: don't do it, you should use https://nodejs.org/api/util.html#util_util_deprecate_fn_msg_code

Comment (snitin315) [reply lib/Server.js]: Not sure how to use `util.deprecate` on the constructor, I used `process.emitWarning` instead - 

Screenshot 2021-07-24 at 9 42 18 AM

Comment (ylemkimon) [reply lib/Server.js]: @snitin315 You can pass an empty function to `util.deprecate()` and call it.

Comment (snitin315) [reply lib/Server.js]: @ylemkimon thanks 👍🏻

**Review by alexander-akait — APPROVED** [review]
I will fix tests today

**Pull request merged**